types/openapi-factory/index.d.ts

```diff
@@ -29,29 +47,31 @@ declare class OpenApi {
     onEvent(onEventFunc: (req?: any) => Promise<any>): void;
     onSchedule(onScheduleFunc: (req?: any) => Promise<any>): void;
 
-    head(route: string, handler: (req?: object) => (OpenApi.HttpResponse | Promise<OpenApi.HttpResponse>)): void;
-    head(route: string, options: OpenApi.HttpMethodOptions, handler: (req?: any) => any): void;
+    /* tslint:disable:max-line-length no-unnecessary-generics */
+    head<RequestBodyType = void, ResponseBodyType = void>(route: string, handler: (req: OpenApi.HttpRequest<RequestBodyType>) => (OpenApi.HttpResponse<ResponseBodyType | null> | Promise<OpenApi.HttpResponse<ResponseBodyType | null>>)): void;
+    head<RequestBodyType = void, ResponseBodyType = void>(route: string, options: OpenApi.HttpMethodOptions, handler: (req: OpenApi.HttpRequest<RequestBodyType>) => (OpenApi.HttpResponse<ResponseBodyType | null> | Promise<OpenApi.HttpResponse<ResponseBodyType | null>>)): void;
 
-    get(route: string, handler: (req?: object) => (OpenApi.HttpResponse | Promise<OpenApi.HttpResponse>)): void;
-    get(route: string, options: OpenApi.HttpMethodOptions, handler: (req?: object) => (OpenApi.HttpResponse | Promise<OpenApi.HttpResponse>)): void;
+    get<RequestBodyType = void, ResponseBodyType = void>(route: string, handler: (req: OpenApi.HttpRequest<RequestBodyType>) => (OpenApi.HttpResponse<ResponseBodyType | null> | Promise<OpenApi.HttpResponse<ResponseBodyType | null>>)): void;
```

Comment:
`void` is definitely not what you want here; that should only be the type of functions that don't return anything. Did you mean `object` or `{}`?

Reply:
In fact, I'm not even sure your usage of `RequestBodyType` here is helping you at all. What are you trying to do with it? See my comment in the test file.